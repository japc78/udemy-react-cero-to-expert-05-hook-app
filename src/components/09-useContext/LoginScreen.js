import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
	// Tarea.
	// 1. Obtener la referencia al context
	// 2. Extrare el setUser
	const { setUser } = useContext(UserContext)

	// 3. Enviar el objeto con setUser
	//  {
	// 	id: 1234,
	// 	name: 'Bitcero',
	// 	email: 'dev@bitcero.com'
	// }

	const newUser =  {
		id: 1234,
		name: 'Bitcero',
		email: 'dev@bitcero.com'
	}

	const handleLogin = () => {
		setUser(newUser)
	}

	return (
		<div>
			<h1>LoginScreen</h1>
			<hr/>

			<button
				className="btn btn-primary"
				onClick={ handleLogin }
			>
					Login
			</button>
		</div>
	)
}
