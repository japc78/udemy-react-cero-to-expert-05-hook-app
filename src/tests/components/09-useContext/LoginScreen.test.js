import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { user } from './fixtures/userData'

describe('Test about UseContext with <LoginScreen />', () => {

	const setUser = jest.fn();

	const wrapper = mount(
		<UserContext.Provider value={{
			setUser
		 }}>
			<LoginScreen />
		</UserContext.Provider>
	);

	test('should be shown correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should run the setUser with the expected argument', () => {
		wrapper.find('button').simulate('click');
		// wrapper.find('button').prop('onClick')();

		//Se expera que el serUser haya sido llamado con ciertos argumentos
		expect( setUser ).toHaveBeenCalledWith(user);
	});
});
