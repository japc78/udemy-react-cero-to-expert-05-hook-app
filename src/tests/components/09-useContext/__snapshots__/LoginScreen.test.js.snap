// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test about UseContext with <LoginScreen /> should be shown correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
